### DISCUSSION TITLE
About width: fit-content

### DISCUSSION THREAD

**Discussion by berkaycubuk** [tailwindlabs/tailwindcss#2727] [Ideas]
Hello dear Tailwind community 👋

In my projects especially in a tags I need to use width: fit-content when I use display: block because I want to have a tag with smallest width possible, so to make it happen I use width: fit-content. In pure CSS is there another way to do that? If not we should add it to Tailwind.

Thanks 🙂

**Comment by joshxyzhimself** [comment]
Can try `w-auto inline-block` as per https://tailwindcss.com/docs/width#auto

**Reply by rotimi-best** [reply]
Thanks, worked like magic

**Reply by danielfoxp2** [reply]
Save the day, thank you.

**Reply by saf-alaeldin** [reply]
Worked! 
Thanks :)

**Reply by vincy2403** [reply]
thank you very much....save the day

**Reply by TheJazzDev** [reply]
Thanks a lot man, it work like charm

**Comment by bertdeblock** [comment]
You could also add the following to your theme?

```javascript
width: {
  'fit-content': 'fit-content'
}
```

**Comment by pumano** [comment]
@adamwathan would love to see `width: fit-content` out of the box. Please add it 👍

**Reply by sshanzel** [reply]
Hoping this will be implemented.

`w-fit` -> `width: fit-content`
`max-w-fit` -> `max-width: fit-content`

The above values are what I used for the extended config.

**Comment by ghost** [comment]
Yep, would like to see this as well for both width and height.

**Comment by dance2die** [comment]
If you have JIT enabled (available since v2.1), you can do `w-[fit-content]`.

**Reply by dalmo3** [reply]
That's a good one, thank you!

**Reply by ShekhSaifuddin007** [reply]
@dance2die how can I enable JIT I'm using `2.0.2`

**Reply by dance2die** [reply]
> @dance2die how can I enable JIT I'm using `2.0.2`

You can try https://github.com/tailwindlabs/tailwindcss-jit, which is what's merged to Tailwind.

**Comment by lukewarlow** [comment]
https://github.com/tailwindlabs/tailwindcss/pull/5638 - I've made a PR to add fit-content to Tailwind

**Reply by lukewarlow** [reply]
It's been merged, so will be in 3.0

**Comment by vasyukovin** [comment]
In some situations "max-w-prose" can handle.

**Comment by RobHowdle** [comment]
Not sure how relevant this is, fairly new with TW. I just created a custom property in my tailwind.config file. Then I just added an `mx-auto` to center it.

`theme: {
    extend: {
        width: {
             'fc': 'fit-content',
         }
    }
}`

**Reply by wongjn** [reply]
`width: fit-content` already exists in Tailwind core as `w-fit`.

**Reply by RobHowdle** [reply]
My mistake. I was looking at the v2 documentation, not v3!

**Comment by Doorkneel** [comment]
Not sure when this was added, couldn't find it in the docs, but "w-fit" works for me.
